lib/Sema/TypeCheckAttr.cpp

```diff
@@ -3008,69 +3009,44 @@ static IndexSubset *computeDifferentiationParameters(
+    GenericEnvironment *transposeGenEnv, SourceLoc attrLoc) {
   // Get function type and parameters.
-  TC.resolveDeclSignature(transposeFunc);
-  auto *functionType = transposeFunc->getInterfaceType()
-                           ->castTo<AnyFunctionType>();
+  TC.resolveDeclSignature(transposeFunction);
+  auto *transposeFunctionType =
+      transposeFunction->getInterfaceType()->castTo<AnyFunctionType>();
   
   ArrayRef<TupleTypeElt> transposeResultTypes;
   // Return type of '@transposing' function can have single type or tuple
   // of types.
-  auto temp = functionType->getResult();
+  auto transposeResultType = transposeFunctionType->getResult();
   if (isCurried)
-    temp = temp->getAs<AnyFunctionType>()->getResult();
-  
-  if (auto t = temp->getAs<TupleType>()) {
-    transposeResultTypes = t->getElements();
+    transposeResultType =
+        transposeResultType->castTo<AnyFunctionType>()->getResult();
+  if (auto resultTupleType = transposeResultType->getAs<TupleType>()) {
+    transposeResultTypes = resultTupleType->getElements();
   } else {
-    transposeResultTypes = ArrayRef<TupleTypeElt>(temp);
+    transposeResultTypes = ArrayRef<TupleTypeElt>(transposeResultType);
   }
-  
-  auto &params = *transposeFunc->getParameters();
-  auto isInstanceMethod = transposeFunc->isInstanceMember();
-  
-  bool wrtSelf = false;
-  if (isCurried && !parsedWrtParams.empty() &&
-      parsedWrtParams.front().getKind() == ParsedAutoDiffParameter::Kind::Self)
-    wrtSelf = true;
-  
-  // Make sure the self type is differentiable.
-  if (isCurried && wrtSelf) {
-    auto selfType = transposeFunc->getImplicitSelfDecl()->getInterfaceType();
-    if (derivativeGenEnv)
-      selfType = derivativeGenEnv->mapTypeIntoContext(selfType);
-    if (!conformsToDifferentiable(selfType, transposeFunc)) {
-      TC.diagnose(attrLoc, diag::diff_function_no_parameters,
-                  transposeFunc->getFullName())
-          .highlight(transposeFunc->getSignatureSourceRange());
-      return nullptr;
-    }
-  }
-  
-  // If parsed differentiation parameters are empty, infer parameter indices
-  // from the function type.
-  // TODO(bartchr): still need to do this!
-//  if (parsedWrtParams.empty())
-//    return TypeChecker::inferTransposingParameters(
-//        function, derivativeGenEnv);
-  
+  auto isInstanceMethod = transposeFunction->isInstanceMember();
+
   // Otherwise, build parameter indices from parsed differentiation parameters.
-  unsigned numParams = params.size() + transposeResultTypes.size();
-  auto paramIndices = SmallBitVector(numParams);
+  auto numUncurriedParams = transposeFunctionType->getNumParams();
+  if (auto *resultFnType =
+      transposeFunctionType->getResult()->getAs<AnyFunctionType>()) {
+    numUncurriedParams += resultFnType->getNumParams();
+  }
+  auto numParams = numUncurriedParams + parsedWrtParams.size() - 1;
+  auto parameterBits = SmallBitVector(numParams);
```

Comment:
Note: `@transposing` attribute parameter indices capacity is fixed here.
Correct capacity: `non-wrt parameter count + wrt parameter count - 1`.